pkg/kubelet/kubelet.go

```diff
@@ -2522,7 +2522,15 @@ func (kl *Kubelet) ListenAndServePodResources() {
 		klog.V(2).InfoS("Failed to get local endpoint for PodResources endpoint", "err", err)
 		return
 	}
-	server.ListenAndServePodResources(socket, kl.podManager, kl.containerManager, kl.containerManager, kl.containerManager)
+
```

Comment:
@moshe010  thanks for the suggestion of splitting structs. I like this direction more. I think it would be impractical for you to wait for this PR to land, so if you also like this direction, feel free to cherry pick this commit (or rewrite it since it's very simple) in your PR

Reply:
Sure, will do :)